src/raw/mod.rs

```diff
@@ -76,26 +216,42 @@ impl<K, V> Table<K, V> {
 
 impl<K, V> Drop for Table<K, V> {
     fn drop(&mut self) {
-        // we need to drop any forwarding nodes (since they are heap allocated).
-
         // safety: we have &mut self _and_ all references we have returned are bound to the
         // lifetime of their borrow of self, so there cannot be any outstanding references to
         // anything in the map.
         let guard = unsafe { crossbeam_epoch::unprotected() };
 
+        // since BinEntry::Nodes are either dropped by drop_bins or transferred to a new table,
+        // all bins are empty or contain a Shared pointing to shared the BinEntry::Moved (if
+        // self.bins was not replaced by drop_bins anyway)
         for bin in &self.bins[..] {
             let bin = bin.swap(Shared::null(), Ordering::SeqCst, guard);
```

Comment:
A separate observation -- this loop is probably quite expensive due to the write here when all we're doing it checking that there aren't any `Node`s in it, and dropping any references. We could probably do better by just swapping out `bins` with an empty `Box<[...]>` instead.

Reply:
Hm, this is done in `drop_bins` but not here, I wonder why. If we decided against the checks (see also above), would we even need to swap anything or would `bins` just get dropped automatically?